libs/wingc/src/lsp/completions.rs

```diff
@@ -144,36 +144,37 @@ pub fn on_completion(params: lsp_types::CompletionParams) -> CompletionResponse
 			if parent.kind() == "custom_type" {
 				if let Some(nearest_type_annotation) = scope_visitor.nearest_type_annotation {
 					if let TypeAnnotationKind::UserDefined(udt) = &nearest_type_annotation.kind {
-						if udt.fields.is_empty() {
-							// this is probably a namespace
-							// `resolve_user_defined_type` will fail for namespaces, let's just look it up instead
+						let type_lookup = resolve_user_defined_type(udt, found_env, scope_visitor.found_stmt_index.unwrap());
+
+						if let Ok(type_lookup) = type_lookup {
+							return get_completions_from_type(&type_lookup, types, Some(found_env.phase), false);
+						} else {
+							// this is probably a namespace, let's look it up
 							let namespace = root_env
-								.lookup_nested_str(&udt.root.name, scope_visitor.found_stmt_index)
+								.lookup_nested_str(&udt.full_path_str(), scope_visitor.found_stmt_index)
 								.ok();
```

Comment:
(optional) There might be enough logic here to warrant moving this else branch into a separate function #nevernester

Reply:
I removed a layer of nesting but otherwise kept it in place